crates/valence_stresser/src/main.rs

```diff
@@ -0,0 +1,32 @@
+use core::time::Duration;
+use std::net::ToSocketAddrs;
+use std::thread::{self, JoinHandle};
+
+use args::StresserArgs;
+use clap::Parser;
+use stresser::make_connection;
+
+mod args;
+pub mod stresser;
+
+fn main() {
+    let args = StresserArgs::parse();
+
+    let target_addr = args.target_host.to_socket_addrs().unwrap().next().unwrap();
+
+    let mut last_thread: Option<JoinHandle<()>> = None;
+
+    for conn_index in 0..args.connections_count {
+        let conn_name = format!("{}{}", args.name_prefix, conn_index);
+
+        last_thread = Some(thread::spawn(move || {
+            make_connection(target_addr, conn_name.as_str());
+        }));
```

Comment:
This is really inefficient and makes it so that the kernel needs to be heavily relied on to manage thread execution. It would be way more preferable to make this async instead.

Reply:
I thought about using an asynchronous way of running connections, but I think that a dedicated thread for each connection would be the most effective way to stress a server, rather than relying on an asynchronous runtime.

Reply:
@dyc3 is right. async should simply improve the performance of the stresser.

Reply:
In general, async programs will out perform threaded programs when there is IO involved, which is very much the case here. Valence's network code is async for this reason. The OS is allowed to switch the executing thread for any given CPU core at any time, which includes interrupting code that is not blocked by IO operations. These context swtiches are not free, not to mention that there is also a memory overhead for each thread.

I tried running the stresser with 4000 clients and a low join cooldown value. My computer seized up for a good 10 seconds because of the threads being created.

Reply:
Alright, I'll research it more, do some benchmarks, and apply the changes.